crates/driver/src/domain/competition/solution/fee.rs

```diff
@@ -85,6 +85,13 @@ impl Fulfillment {
                 tracing::debug!(uid=?self.order().uid, fee_from_surplus=?fee_from_surplus, fee_from_volume=?fee_from_volume, protocol_fee=?(std::cmp::min(fee_from_surplus, fee_from_volume)), executed=?self.executed(), surplus_fee=?self.surplus_fee(), "calculated protocol fee");
                 Ok(std::cmp::min(fee_from_surplus, fee_from_volume))
             }
+            Some(FeePolicy::PriceImprovement {
+                factor: _,
+                max_volume_factor: _,
+                quote: _,
+            }) => {
+                todo!()
```

Comment:
How quote should be used here?

Reply:
I think the cleanest solution would probably to adjust `fee_from_surplus()` such that it computes the surplus compared to a given reference price. Then you could pass in the user signed limit prices for `FeePolicy::Surplus` or the quoted amounts for `FeePolicy::PriceImprovement`.

Reply:
Let me know if I got it wrong https://github.com/cowprotocol/services/pull/2277/commits/4a0e5ff93621d5601937b0d67bc6e703c1e6a27f